internal/cri/sputil/seccomp_linux_test.go

```diff
@@ -0,0 +1,228 @@
+			disable: true,
+			sp: &runtime.SecurityProfile{
+				ProfileType: runtime.SecurityProfile_Unconfined,
+			},
+		},
+		{
+			desc:       "sp should not set seccomp when privileged is true",
+			privileged: true,
+			sp: &runtime.SecurityProfile{
+				ProfileType: runtime.SecurityProfile_RuntimeDefault,
+			},
+		},
+		{
+			desc: "sp should not set seccomp when seccomp is unconfined",
+			sp: &runtime.SecurityProfile{
+				ProfileType: runtime.SecurityProfile_Unconfined,
+			},
+		},
+		{
+			desc: "sp should not set seccomp when seccomp is not specified",
+		},
+		{
+			desc:     "sp should set default seccomp when seccomp is runtime/default",
+			specOpts: seccomp.WithDefaultProfile(),
+			sp: &runtime.SecurityProfile{
+				ProfileType: runtime.SecurityProfile_RuntimeDefault,
+			},
+		},
+		{
+			desc:     "sp should set specified profile when local profile is specified",
+			specOpts: seccomp.WithProfile("test-profile"),
+			sp: &runtime.SecurityProfile{
+				ProfileType:  runtime.SecurityProfile_Localhost,
+				LocalhostRef: profileNamePrefix + "test-profile",
+			},
+		},
+		{
+			desc:     "sp should set specified profile when local profile is specified even without prefix",
+			specOpts: seccomp.WithProfile("test-profile"),
+			sp: &runtime.SecurityProfile{
+				ProfileType:  runtime.SecurityProfile_Localhost,
+				LocalhostRef: "test-profile",
+			},
+		},
+		{
+			desc:      "sp should return error if specified profile is invalid",
+			expectErr: true,
+			sp: &runtime.SecurityProfile{
+				ProfileType:  runtime.SecurityProfile_RuntimeDefault,
+				LocalhostRef: "test-profile",
+			},
+		},
+	} {
+		t.Run(test.desc, func(t *testing.T) {
+			ssp := test.sp
+			csp, err := GenerateSeccompSecurityProfile(
+				test.profile,
+				test.defaultProfile)
+			if err != nil {
+				if test.expectErr {
+					assert.Error(t, err)
+				} else {
+					assert.NoError(t, err)
+				}
+			} else {
+				if ssp == nil {
+					ssp = csp
+				}
+				specOpts, err := GenerateSeccompSpecOpts(ssp, test.privileged, !test.disable)
+				if test.expectErr {
+					assert.Error(t, err)
+				} else {
+					assert.NoError(t, err)
+					if test.specOpts == nil && specOpts == nil {
+						return
+					}
+					if test.specOpts == nil || specOpts == nil {
+						t.Fatalf("unexpected nil specOpts, expected nil: %t, actual nil: %t", test.specOpts == nil, specOpts == nil)
+					}
+					// `specOpts` for seccomp only uses/modifies `*specs.Spec`, not
+					// `oci.Client` or `*containers.Container`, so let's construct a
+					// `*specs.Spec` and compare if the results are the same.
+					expected := runtimespec.Spec{
+						Linux: &runtimespec.Linux{},
+						Process: &runtimespec.Process{
+							Capabilities: &runtimespec.LinuxCapabilities{
+								// This is to ensure the test covers logic in
+								// `contrib/seccomp.DefaultProfile` as much as possible
+								Bounding: []string{
```

Comment:
Hi @fuweid, regarding this previous comment in a separate PR (https://github.com/containerd/containerd/pull/11264/files#r1955045128), the intention is to cover the logic in `contrib/seccomp.DefaultProfile` (added a comment). Let me know if you prefer just removing this, I can do it as part of this refactor PR. Thanks!

> I think we don't need this assignment here. It's kind of confusing.
Anyway, we can do it in the follow-up.

https://github.com/containerd/containerd/blob/6f652853f01ef9ba340a860c2f39edf1701102d1/contrib/seccomp/seccomp_default.go#L568